presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketing.java

```diff
@@ -287,7 +305,8 @@ private static int hashBytes(int initialValue, Slice bytes)
 
     public static Optional<HiveBucketFilter> getHiveBucketFilter(Table table, TupleDomain<ColumnHandle> effectivePredicate)
     {
-        if (!table.getStorage().getBucketProperty().isPresent()) {
+        Optional<HiveBucketProperty> hiveBucketProperty = table.getStorage().getBucketProperty();
+        if (!hiveBucketProperty.isPresent() || !hiveBucketProperty.get().getBucketFunctionType().equals(HIVE_COMPATIBLE)) {
```

Comment:
This one is needed to return empty HiveBucketFilter:   ignore bucket pruning

Reply:
I'm not sure if it is safe to ignore it. There shouldn't be any bucket pruning for temporary tables though.

Reply:
Since code path exist for bucketed temporary table : `Optional<HiveBucketFilter> bucketFilter = shouldIgnoreTableBucketing ? Optional.empty() : getHiveBucketFilter(table, effectivePredicate);`    Thus we cannot use assertion but should ignore.

Reply:
Or add the check in call site: `shouldIgnoreTableBucketing || bucket function type == PRESTO_NATIVE ? Optional.empty() : getHiveBucketFilter(table, effectivePredicate);`
Same thing, let's keep it inside.

Reply:
It feels like this shortcut is not needed. Temporary tables are never filtered. Thus it seems like the method should return at `if (!bindings.isPresent()) {`, before reaching the `getHiveBuckets(table, bindings.get())` that would fail if the hash function is not `HIVE_COMPATIBLE`

Reply:
`bindings` is a non-empty Optional object for temporary table, query will fail if I revert this change. The shortcut has its value to avoid unneeded code path.

Reply:
Okay, I just checked the code, and it seems to be safe to skip bucket filter.

Let's at least extract this condition into a separate branch and add a comment, e.g.:

```
if (!hiveBucketProperty.isPresent()) {
  return Optional.empty();
}
if(!hiveBucketProperty.get().getBucketFunctionType().equals(HIVE_COMPATIBLE)){
  // bucket filtering is only supported for tables bucketed with HIVE_COMPATIBLE hash function
  return Optional.empty();
}
```